src/wp-includes/class-wp-user-query.php

```diff
@@ -842,19 +840,19 @@ public function query() {
 			foreach ( $this->results as $result ) {
 				$result->id = $result->ID;
 			}
-		} elseif ( 'all_with_meta' === $qv['fields'] ) {
+		} elseif ( 'all_with_meta' === $qv['fields'] || 'all' === $qv['fields'] ) {
 			cache_users( $this->results );
 
 			$r = array();
 			foreach ( $this->results as $userid ) {
-				$r[ $userid ] = new WP_User( $userid, '', $qv['blog_id'] );
+				if ( 'all_with_meta' === $qv['fields'] ) {
+					$r[ $userid ] = new WP_User( $userid, '', $qv['blog_id'] );
+				} else {
+					$r[] = new WP_User( $userid, '', $qv['blog_id'] );
+				}
```

Comment:
Is this the code that prevents you from normalising `all_with_fields` to `all` at the start of the function?

Reply:
I am not sure of the question. This code ensure that `all_with_meta` returns an array of with key of user id. Whereas `all` return an array of WP_User objects with a numeric key.

Reply:
> This code ensure that `all_with_meta` returns an array of with key of user id. Whereas `all` return an array of WP_User objects with a numeric key.

That answers my question, thanks.